packages/compose/src/hooks/use-constrained-tabbing/index.js

```diff
@@ -62,9 +62,11 @@ function useConstrainedTabbing() {
 
 			trap.tabIndex = -1;
 			node[ domAction ]( trap );
+
+			// Remove itself when the trap loses focus.
+			trap.addEventListener( 'blur', () => node.removeChild( trap ) );
+
 			trap.focus();
-			// Remove after the browser moves focus to the next element.
-			timeoutId = setTimeout( () => node.removeChild( trap ) );
```

Comment:
Since this timeout is not used any longer, the variable declaration at line 36 and the clearTimeout at line 75 should be removed as well.